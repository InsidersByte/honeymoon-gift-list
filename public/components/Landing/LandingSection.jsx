import React from 'react';

import './LandingSection.styl';

export default function LandingSection(props) {
    return (
        <section className="landing-section">
            <h1 className="landing-section__title">{props.title} <span></span></h1>

            <div className="landing-section__content">
                {props.children}
            </div>

            {props.postContent}
        </section>
    );
}

LandingSection.propTypes = {
    children: React.PropTypes.element,
    postContent: React.PropTypes.element,
    title: React.PropTypes.string.isRequired,
};
