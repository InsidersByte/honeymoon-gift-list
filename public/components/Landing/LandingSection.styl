.root
    padding 50px 0
    text-align center
    color rgb(154, 156, 167)

    h1, h2, h3, h4, h5, h6
        color rgb(82, 105, 120)

    a
        color rgb(0, 173, 237)
        text-decoration underline

        &:hover
            color rgb(0, 173, 237)

    &:nth-child(even)
        color white
        background #00aded

        h1, h2, h3, h4, h5, h6
            color white

        a
            color white

            &:hover
                color white

    &:before
        display table
        content ' '

    &:after
        display table
        content ' '
        clear both

    &:last-of-type
        padding-bottom 0

.content
    padding-top 10px
    font-size 0.9em
    margin 0 25%

    img
        width 100%
        max-width 500px
        height auto

.title
    text-transform capitalize
    text-align center
    font-size 2.5em

@media (max-width 736px)
    .content
        margin 0 5%
