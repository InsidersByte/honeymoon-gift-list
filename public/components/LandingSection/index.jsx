/* @flow */

import React, { Component } from 'react';
import MarkdownRenderer from 'react-markdown-renderer';
import LandingItem from '../LandingItem';

type PropsType = {
    title: string,
    content: string,
};

export default class LandingSection extends Component<void, PropsType, void> {
    landingItem: LandingItem;

    scrollTo = () => {
        const landingItem = this.landingItem;
        landingItem.scrollTo();
    };

    render() {
        const { title, content } = this.props;

        return (
            <LandingItem title={title} ref={(c) => { this.landingItem = c; }}>
                <MarkdownRenderer markdown={content} />
            </LandingItem>
        );
    }
}
