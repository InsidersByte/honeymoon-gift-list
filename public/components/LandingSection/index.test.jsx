import React from 'react';
import { shallow, mount } from 'enzyme';
import smoothscroll from 'smoothscroll';
import LandingSection from './';

jest.mock('smoothscroll');

describe('LandingItem', () => {
  it('should render correctly', () => {
    const title = 'Hello World';
    const content = 'I am some content';

    const wrapper = shallow(<LandingSection title={title} content={content} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call smoothscroll when calling scrollTo', () => {
    const title = 'Hello World';
    const content = 'I am some content';

    const wrapper = mount(<LandingSection title={title} content={content} />);

    expect(wrapper.instance().scrollTo).toBeInstanceOf(Function);
    expect(smoothscroll).not.toHaveBeenCalled();

    wrapper.instance().scrollTo();

    expect(smoothscroll).toHaveBeenCalledTimes(1);
    expect(smoothscroll).toHaveBeenCalledWith(wrapper.instance().landingItem.container);
  });
});
