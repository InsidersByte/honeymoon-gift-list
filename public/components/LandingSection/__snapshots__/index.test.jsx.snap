// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingItem should render correctly 1`] = `
<LandingItem
  title="Hello World"
>
  <MarkdownRenderer
    markdown="I am some content"
  />
</LandingItem>
`;
